src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -333,6 +333,15 @@ public bool GetUnificationUseSiteDiagnosticRecursive(ref DiagnosticInfo result,
 
         public void CheckAllConstraints(ConversionsBase conversions, Location location, DiagnosticBag diagnostics)
         {
+            // The first unconstrained type parameter in the type.
+            var typeParameter = this.VisitType(
```

Comment:
Note: This code path only handles a fraction of the cases where `T?` can appear. The diagnostic probably needs to be reported elsewhere.